components/central-application-gateway/README.md

```diff
@@ -0,0 +1,109 @@
+- **skipVerify** is the flag for skipping the verification of certificates for the proxy targets. The default value is `false`.
+- **requestLogging** is the flag for logging incoming requests. The default value is `false`.
+- **proxyTimeout** is the timeout for requests sent through the proxy, expressed in seconds. The default value is `10`.
+- **proxyCacheTTL** is the time to live of the remote API information stored in the proxy cache, expressed in seconds. The default value is `120`.
+
+
+## API
+The Central Application Gateway exposes:
+- external API implementing health endpoint for liveness and readiness probes
+- internal API implementing proxy handler accessible via a service of ClusterIP type
+
+### Standalone mode
+In case  **disableLegacyConnectivity** is `false` the proxy API exposes the following endpoint:
+```{Application name}/{Service name}/{target API path}``` 
+
+For instance, if the user registered `cc-occ-commerce-webservices` service in `ec` application using Application Registry, they can send a request to the following url:
+```http://central-application-gateway:8080/ec/cc-occ-commerce-webservices/basesites```
+
+As a result Central Application Gateway performs the following steps:
+- looks for `cc-occ-commerce-webservices` service in `ec` Application CRD and extracts target URL path along with authentication configuration
+- modifies request to include authentication data
+- sends request to the following path `{target URL extracted from the Application CRD}/basesites`
+
+### Compass mode
+In case  **disableLegacyConnectivity** is `true` the proxy API exposes the following endpoint:
+```{Application name}/{API bundle name}/{API definition name}/{target api path}``` 
+
+For instance, if the user registered `cc-occ` API bundle with `commerce-webservices` API definition in `ec` application, they can send a request to the following url:
+```http://central-application-gateway:8082/ec/cc-occ/commerce-webservices/basesites``` 
+
+As a result Central Application Gateway performs the following steps:
+- looks for `cc-occ` service and `commerce-webservices` entry in `ec` Application CRD and extracts target URL path along with authentication configuration
+- modifies request to include authentication data
+- sends request to the following path `{target URL extracted from the Application CRD}/basesites`
+
+## Development
+
+This section explains the development process.
+
+### Generate mocks
+
+Prerequisites:
+
+ - [Mockery](https://github.com/vektra/mockery) 2.0 or higher
+
+To generate mocks, run:
+
+```sh
+go generate ./...
+```
+
+When adding a new interface to be mocked or when a mock of an existing interface is not being generated, add the following line directly above the interface declaration:
+
+```
+//go:generate mockery --name {INTERFACE_NAME}
+```
+
+### Tests
+
+This section outlines the testing details.
+
+#### Unit tests
+
+To run the unit tests, use the following command:
```

Comment:
```suggestion
To run the unit tests, run this command:
```
It's not a mistake, but we should strive for simplicity whenever we can. And also, our [guidelines](https://github.com/kyma-project/community/blob/main/guidelines/content-guidelines/08-style-and-terminology.md#terminology) say to use "run this command" with commands.